app/models/king.rb

```diff
@@ -1,9 +1,37 @@
+require 'pry'
 class King < Piece
   # KINGS CAN MOVE ONE SQUARE IN ANY DIRECTION
 
   def valid_move?(destination_x, destination_y)
     return false if super(destination_x, destination_y) == false
-    (self.current_x - destination_x).abs <= 1 && 
+    self.castling?(destination_x, destination_y) ||
+    	(self.current_x - destination_x).abs <= 1 && 
          (self.current_y - destination_y).abs <= 1
   end
-end
\ No newline at end of file
+
+  def castling?(destination_x, destination_y)
+   !self.has_moved? && !rook_on_side_moved?(destination_x, destination_y) &&
+   		!pieces_between?(destination_x, destination_y)
+  end
+
+  def rook_on_side_moved?(destination_x, destination_y)

```

Comment:
Are there any other rooks besides those that are on the side? Why not just name it `rook_moved?`